adr/0010-release-process.md

```diff
@@ -0,0 +1,25 @@
+# Release Process
+
+Date: 2018-12-03
+
+## Status
+
+Draft
+
+## Context
+
+eLife has some projects that are published as packages, e.g. for Python projects on pypi.
```

Comment:
Not sure if we should try to make this ADR language agnostic or whether it's better to stick to Python here. Java for example has a more defined release process (e.g. including SNAPSHOT versions).